### PR TITLE
Add TrioConverter, use custom tf split function.

### PR DESCRIPTION
**Pull request by adarob**

### CODE REVIEW

File: core/src/data.ts [hunk 1]
```diff
@@ -77,9 +81,10 @@ export function converterFromSpec(spec: ConverterSpec) {
  * `NoteSequence` objects.
  */
 export abstract class DataConverter {
-  abstract numSteps: number;  // Total length of sequences.
-  abstract numSegments: number;  // Number of steps for conductor.
-  abstract readonly NUM_SPLITS: number;  // Number of conductor splits.
+  abstract readonly numSteps: number;  // Total length of sequences.
+  abstract readonly numSegments: number;  // Number of steps for conductor.
+  abstract readonly depth: number;  // Size of final output dimension.
+  abstract readonly NUM_SPLITS: number;  // Const number of conductor splits.
```
Comment (cghawthorne) [inline core/src/data.ts]: now that the other properties are also readonly, should they either all be uppercase or should NUM_SPLITS be lowercase?

File: core/src/sequences.ts [hunk 2]
```diff
@@ -74,6 +74,10 @@ export class Sequences {
     return n && (n & (n - 1)) === 0;
   }
 
+  public static clone(ns:INoteSequence) {
```
Comment (cghawthorne) [inline core/src/sequences.ts]: Thanks, makes sense to centralize this logic. Could you also update sequences_test.ts to use this in the few places it clones NoteSequences?

File: core/src/tf_lib.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as tf from '@tensorflow/tfjs';
+
+export function split(
```
Comment (cghawthorne) [inline core/src/tf_lib.ts]: is this something we should contribute back to tfjs?

File: core/src/data.ts (hunk 1 above)
Comment (adarob) [reply core/src/data.ts]: No, NUM_SPLITS is still something different. It really should be const but const isn't allowed in the abstract definition...

File: core/src/tf_lib.ts (hunk 3 above)
Comment (adarob) [reply core/src/tf_lib.ts]: Working on it now :)

Comment (adarob) [reply core/src/tf_lib.ts]: https://github.com/tensorflow/tfjs-core/pull/930

**Comment by adarob** [comment]
#9

**Pull request merged**